File: app/src/main/java/soup/movie/data/MovieListResponse.java
package soup.movie.data;

public class MovieListResponse {

    private MovieListResult movieListResult;

    public MovieListResult getMovieListResult() {
        return movieListResult;
    }

    @Override
    public String toString() {
        return "MovieListResponse{" +
                "movieListResult=" + movieListResult +
                '}';
    }
}

File: data/src/main/java/soup/movie/data/model/response/MovieListResponse.kt
package soup.movie.data.model.response

import soup.movie.data.model.Movie
import soup.movie.data.model.response.CachedMovieList.Companion.STALE_MS

data class MovieListResponse(
    val lastUpdateTime: Long,
    val list: List<Movie>
)

fun MovieListResponse.isStaleness(): Boolean {
    return (list.isEmpty() || System.currentTimeMillis() - lastUpdateTime > STALE_MS)
}
